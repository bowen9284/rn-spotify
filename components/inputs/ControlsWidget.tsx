import React, { useContext, useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { AuthContext } from '../../context/authContext';

interface Props {
  isFollowed: boolean;
  onFollowPress: (toggle: boolean) => void;
  onDownloadPress: () => void;
  onEllipsisPress: () => void;
}
const ControlsWidget: React.FC<Props> = ({
  isFollowed,
  onFollowPress,
  onDownloadPress,
  onEllipsisPress,
}) => {
  const context = useContext(AuthContext);

  let isFollowedIcon = () => {
    return isFollowed ? (
      <Ionicons name="ios-heart" size={30} color="green" />
    ) : (
      <Ionicons name="heart-outline" size={30} color="gray" />
    );
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={() => {
          onFollowPress(!isFollowed);
        }}
      >
        {isFollowedIcon()}
      </TouchableOpacity>
      <TouchableOpacity onPress={() => {}}>
        <Ionicons name="arrow-down-circle-outline" size={30} color="gray" />
      </TouchableOpacity>
      <TouchableOpacity onPress={() => {}}>
        <Ionicons name="ellipsis-horizontal" size={30} color="gray" />
      </TouchableOpacity>
    </View>
  );
};

export default ControlsWidget;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
});
