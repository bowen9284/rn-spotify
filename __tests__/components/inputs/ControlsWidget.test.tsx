import React from "react";
import renderer from "react-test-renderer";
import ControlsWidget from "../../../components/inputs/ControlsWidget";


const props = {
  description: 'Description about the playlist',
  isFollowed: true,
  ownerDisplayName: "Spotify",
  ownerImages: [{
    width: 640,
    height: 640,
    url: 'someUrl' 
  }],
  numOfLikes: 200000,
  totalDuration: 415262,
  onFollowPress: jest.fn(),
  onDownloadPress: jest.fn(),
  onEllipsisPress: jest.fn()
};

test("renders correctly", () => {
  const tree = renderer.create(<ControlsWidget {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
