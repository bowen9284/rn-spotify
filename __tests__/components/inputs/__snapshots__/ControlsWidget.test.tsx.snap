// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "color": "lightgray",
          "fontSize": 13,
        }
      }
    >
      Description about the playlist
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 15,
      }
    }
  >
    <View
      style={
        Object {
          "width": "45%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(28, 28, 30)",
            "fontWeight": "bold",
          }
        }
      />
      <Text
        ellipsizeMode="tail"
        style={
          Object {
            "color": "lightgray",
            "fontSize": 13,
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        200,000
         likes Â·
         
        19h 6m
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "flex-end",
          "width": "55%",
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
